### ISSUE TITLE
Bug: Crashes with open loops

### ISSUE THREAD

**Issue by CSChisholm** [shredEngineer/MagnetiCalc#3]
**Description**: When calculating the field with loops where the end point is not the same as the start point, the program crashes just after the calculation finishes
```
WARNING: Traceback (most recent call last):
  File "<filter>/.local/lib/python3.7/site-packages/magneticalc/CalculationThread.py", line 180, in on_parameters_valid
    self.gui.model.on_parameters_valid()
  File "<filter>/.local/lib/python3.7/site-packages/magneticalc/Model.py", line 269, in on_parameters_valid
    self.gui.sidebar_right.parameters_widget.update_labels()
  File "<filter>/.local/lib/python3.7/site-packages/magneticalc/Parameters_Widget.py", line 154, in update_labels
    self.energy_units_label.setText(energy_value.split(" ")[1])
IndexError: list index out of range
Aborted (core dumped)
```
The problem is coming from `update_labels` in `Parameters_Widget.py`. Perhaps changing line 154 to `self.energy_units_label.setText(energy_value.split(" ")[-1])` would fix it. Otherwise a try-catch might be required.

**Platform**:
```
Kernel: 5.4.0-91-generic x86_64
bits: 64
compiler: gcc v: 7.5.0
Desktop: Cinnamon 4.4.8
Distro: Linux Mint 19.3 Tricia base: Ubuntu 18.04 bionic
Python 3.7.4
```
Also tested (less thoroughly) with Windows 10 and Python 3.8. Both installed via pip.

**Minimal example**
```
[DEFAULT]
version = MagnetiCalc v1.9
backend = 0
auto_calculation = True
num_cores = 0
wire_points_base = -0.500000, +0.000000, +0.000000; +0.500000, +0.000000, +0.000000
wire_stretch = 0.1000, 1.0000, 1.0000
wire_slicer_limit = 0.0500
wire_dc = 1.0000
rotational_symmetry_count = 30
rotational_symmetry_radius = 1.0000
rotational_symmetry_axis = 2
rotational_symmetry_offset = 0
sampling_volume_padding = 1, 1, 1
sampling_volume_resolution_exponent = 3
sampling_volume_label_resolution_exponent = 0
field_type = 1
field_distance_limit = 0.0008
color_metric = Log Magnitude
alpha_metric = Magnitude
field_point_scale = 0.0000
field_arrow_head_scale = 0.7500
field_arrow_line_scale = 0.7500
field_boost = 0.0000
display_field_magnitude_labels = True
show_wire_segments = True
show_wire_points = True
show_colored_labels = True
show_coordinate_system = True
show_perspective_info = True
dark_background = True
azimuth = 135
elevation = 35.264389682754654
scale_factor = 3.0000
constraint_count = 0

[User]
auto_calculation = False
rotational_symmetry_count = 1
rotational_symmetry_radius = 0.0000
rotational_symmetry_axis = 2
rotational_symmetry_offset = 0.0000
wire_stretch = +1.000000, +1.000000, +1.000000
```

**Comment by shredEngineer** [comment]
Thank you for reporting this error! I already encountered this, too, and it will be fixed in the upcoming release.

This is not really due to the loop not being closed, but rather due to the fact that your wire geometry is completely one-dimensional.
Subsequently, the calculated energy of the loop is extremely small. In this case, it is 40.8 * 10^-42 J (almost zero, but with numerical noise).

This number is formatted using the si-prefix package, which outputs a string consisting of a value and an SI prefix, separated by a space.
Currently, my code relied on this space for splitting value and prefix, for the purpose of nice visual aligning in PyQt5.

However, for numbers with exponents greater than 24, the the si-prefix package internally switches the format to scientific notation, i.e. "40.8e-42", where the space is missing.
This corner case is now handled by explicitly enforcing a format string which does contain a space character.

**Comment by CSChisholm** [comment]
Great

**Comment by shredEngineer** [comment]
This issue has been fixed in 28d43f9fa365e80a67e78d79947b947d23a443b7.
